### PR TITLE
Fix devfile 2.0.0 and 2.1.0 versions & add 2.2.0-alpha

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Devfile API 2.1.0 is already released so I'm moving 2.1.0-alpha to be just 2.1.0.
Also, until https://github.com/devfile/api/issues/475 is not fixed on Devfile API side, I'm fixing schemaVersion on Che Server side, to allow Dashboard to pick the right schema version to validate devfile.

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/19733

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by sparkoo** [comment]
can you please add tests for 2.2.0-alpha validation? Like the tests for 2.1.0

**Comment by sleshchenko** [comment]
I tried and failed, I appreciate if you add them after my PR is merged, or feel free to push here.

**Review by skabashnyuk — DISMISSED** [review]

**Comment by skabashnyuk** [comment]
>can you please add tests for 2.2.0-alpha validation? Like the tests for 2.1.0

@sparkoo  can you elaborate on that. What did you mean?

**Comment by sparkoo** [comment]
> > can you please add tests for 2.2.0-alpha validation? Like the tests for 2.1.0
> 
> @sparkoo can you elaborate on that. What did you mean?

I mean to add new test devfile samples here https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/devfile/validator/DevfileSchemaValidatorTest.java#L87 and here https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/devfile/validator/DevfileSchemaValidatorTest.java#L277

File: wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_simple-devfile.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-schemaVersion: 2.1.0-alpha
+schemaVersion: 2.1.0
```
Comment (sparkoo) [inline wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_simple-devfile.yaml:13]: file is still named 2-1-0-alpha

File: wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_just_schemaVersion.yaml [hunk 2]
```diff
@@ -11,4 +11,4 @@
 #
 
 ---
-schemaVersion: 2.1.0-alpha
+schemaVersion: 2.1.0
```
Comment (sparkoo) [inline wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_just_schemaVersion.yaml:14]: file is still named 2-1-0-alpha

File: wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_with_invalid_plugin_definition.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@
 #   Red Hat, Inc. - initial API and implementation
 #
 
-schemaVersion: 2.1.0-alpha
+schemaVersion: 2.1.0
```
Comment (sparkoo) [inline wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_with_invalid_plugin_definition.yaml:13]: file is still named 2-1-0-alpha

File: wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/devfile/validator/DevfileSchemaValidatorTest.java [hunk 4]
```diff
@@ -263,15 +263,15 @@ public void shouldThrowExceptionWhenDevfileHasUnsupportedApiVersion() throws Exc
...
-        "Version 'a.b.c' of the devfile is not supported. Supported versions are '[1.0.0, 2.0.0, 2.1.0-alpha]'."
+        "Version 'a.b.c' of the devfile is not supported. Supported versions are '[1.0.0, 2.0.0, 2.1.0, 2.2.0-alpha]'."
       },
       {
         "devfile/devfile_v2_unsupported_schemaVersion.yaml",
-        "Version '22.33.44' of the devfile is not supported. Supported versions are '[1.0.0, 2.0.0, 2.1.0-alpha]'."
+        "Version '22.33.44' of the devfile is not supported. Supported versions are '[1.0.0, 2.0.0, 2.1.0, 2.2.0-alpha]'."
       },
       {
         "devfile/devfile_v2-1-0-alpha_unsupported_schemaVersion.yaml",
```
Comment (sparkoo) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/devfile/validator/DevfileSchemaValidatorTest.java:273]: file is still named 2-1-0-alpha

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Review by sparkoo — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

File: wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_just_schemaVersion.yaml (hunk 2 above)
Comment (sleshchenko) [reply wsmaster/che-core-api-workspace/src/test/resources/devfile/schema_test/devfile/devfile_v2-1-0-alpha_just_schemaVersion.yaml:14]: sorry for missing that. BTW we'll discuss on Monday if we really should use -alpha prefix, so I hope we can avoid adapting devfiles after release.

**Pull request merged**